variant_to_realignment.nf

```diff
@@ -229,13 +349,16 @@ workflow process_split_reads {
         alignWithMinimap(
             extractVariantInfoToFastaHeader.out.variant_read1_with_info,
             extractVariantInfoToFastaHeader.out.variant_read2_with_info,
-            new_genome_fa
+            new_genome_fa, flank_length
         )
-        readsToRemappedVariants(alignWithMinimap.out.reads_aligned_bam, new_genome_fa)
+        sortByName(alignWithMinimap.out.reads_aligned_bam)
+        // This last step will use all alignments enven the one that have alternative secondary alignments
+        readsToRemappedVariants(sortByName.out.reads_aligned_sorted_bam, new_genome_fa, flank_length, false)
```

Comment:
These three workflows (`process_split_reads`, `mid` and `long`) are identical except for `flank_length` and this call to `readsToRemappedVariants`, is that right?  I might prefer to extract these as parameters and reuse the code, unless you think they might diverge more substantially in the future.

Reply:
This is due to a limitation of the Nextflow DSL2.
The signature of a process is define by the workflow that contains it and the process name. This means that they need to be unique and a process can only be called once per workflow.
The only way to call the same process multiple time is to embed it in separate workflow. Hence the three quasi identical workflow. 
See the warning [here](https://www.nextflow.io/docs/latest/dsl2.html#workflow-definition) and note [there](https://www.nextflow.io/docs/latest/dsl2.html#workflow-composition)

Reply:
Got it, thanks for explaining!  DSL2 is taking some getting used to for me...

Reply:
Would something like this work?
* Create a workflow grouping together all of the repeating steps (`convertVCFToBed`, `flankingRegionBed`, `flankingRegionFasta`, ...);
* In each of the three remaining separate workflows (`process_split_reads`, `process_split_reads_mid`, `process_split_reads_long`), call that single workflow with different parameters instead of calling all of the commands separately.

This would reduce the code duplication significantly, while still adhering to the DSL2 principles of not calling the same workflow twice from the same workflow.

Reply:
This does work indeed. I've refactor following your suggestion